Marlin/src/pins/samd/pins_RAMPS_144.h

```diff
@@ -0,0 +1,547 @@
+#endif
+
+#define E0_STEP_PIN         36
+#define E0_DIR_PIN          34
+#define E0_ENABLE_PIN       30
+#ifndef E0_CS_PIN
+  #define E0_CS_PIN         22
+#endif
+
+#define E1_STEP_PIN         26
+#define E1_DIR_PIN          28
+#define E1_ENABLE_PIN       24
+#ifndef E1_CS_PIN
+  #define E1_CS_PIN         43
+#endif
+
+//
+// Temperature Sensors
+//
+#define TEMP_0_PIN          13
+#define TEMP_BED_PIN        14
+#define TEMP_CHAMBER_PIN    15
+
+//
+// Heaters / Fans
+//
+#define HEATER_0_PIN        10
+#define HEATER_BED_PIN      8
+#define FAN_PIN             9
+#define FAN1_PIN            7
+#define FAN2_PIN            12
+
+//
+// Misc. Functions
+//
+#define SDSS                ?53
+#define LED_PIN             13
+
+#ifndef FILWIDTH_PIN
+  #define FILWIDTH_PIN     ?5   // Analog Input on AUX2
+#endif
+
+// RAMPS 1.4.4 DIO 4 on the servos connector
+#ifndef FIL_RUNOUT_PIN
+  #define FIL_RUNOUT_PIN   ?4
+#endif
+
+#ifndef PS_ON_PIN
+  #define PS_ON_PIN        39
+#endif
+
+#if ENABLED(CASE_LIGHT_ENABLE) && !defined(CASE_LIGHT_PIN) && !defined(SPINDLE_LASER_ENA_PIN)
+  #if NUM_SERVOS <= 1 // Prefer the servo connector
+    #define CASE_LIGHT_PIN  6   // Hardware PWM
+  #endif
+#endif
+
+//
+// M3/M4/M5 - Spindle/Laser Control
+//
+#if HAS_CUTTER && !defined(SPINDLE_LASER_ENA_PIN)
+  #if !NUM_SERVOS                         // Use servo connector if possible
+    #define SPINDLE_LASER_ENA_PIN     4   // Pullup or pulldown!
+    #define SPINDLE_LASER_PWM_PIN     6   // Hardware PWM
+    #define SPINDLE_DIR_PIN           5
+  #else
+    #error "No auto-assignable Spindle/Laser pins available."
+  #endif
+#endif
+
+//
+// TMC software SPI
+//
+#if ENABLED(TMC_USE_SW_SPI)
+  #ifndef TMC_SW_MOSI
+    #define TMC_SW_MOSI    ?66
+  #endif
+  #ifndef TMC_SW_MISO
+    #define TMC_SW_MISO    ?44
+  #endif
+  #ifndef TMC_SW_SCK
+    #define TMC_SW_SCK     ?64
+  #endif
+#endif
+
+#if HAS_TMC220x
+  /**
+   * TMC2208/TMC2209 stepper drivers
+   *
+   * Hardware serial communication ports.
+   * If undefined software serial is used according to the pins below
+   */
+  //#define X_HARDWARE_SERIAL  Serial1
+  //#define X2_HARDWARE_SERIAL Serial1
+  //#define Y_HARDWARE_SERIAL  Serial1
+  //#define Y2_HARDWARE_SERIAL Serial1
+  //#define Z_HARDWARE_SERIAL  Serial1
+  //#define Z2_HARDWARE_SERIAL Serial1
+  //#define E0_HARDWARE_SERIAL Serial1
+  //#define E1_HARDWARE_SERIAL Serial1
+  //#define E2_HARDWARE_SERIAL Serial1
+  //#define E3_HARDWARE_SERIAL Serial1
+  //#define E4_HARDWARE_SERIAL Serial1
+
+  //
+  // Software serial
+  //
+
+  #ifndef X_SERIAL_TX_PIN
+    #define X_SERIAL_TX_PIN  ?40
+  #endif
+  #ifndef X_SERIAL_RX_PIN
+    #define X_SERIAL_RX_PIN  ?63
+  #endif
+  #ifndef X2_SERIAL_TX_PIN
+    #define X2_SERIAL_TX_PIN -1
+  #endif
+  #ifndef X2_SERIAL_RX_PIN
+    #define X2_SERIAL_RX_PIN -1
+  #endif
+
+  #ifndef Y_SERIAL_TX_PIN
+    #define Y_SERIAL_TX_PIN  ?59
+  #endif
+  #ifndef Y_SERIAL_RX_PIN
+    #define Y_SERIAL_RX_PIN  ?64
```

Comment:
I don't think we can leave these `?` characters in the file.

Reply:
I know, this is why I asked to @MrAlvin to take a look to this. I don't have full board schematic and I need him help